src/components/DataExplorerApp/VizSettingsForm/makeDefaultVizConfig.tsx

```diff
@@ -1,32 +1,35 @@
 import { match } from "ts-pattern";
-import { VizType } from ".";
 import { BarChartSettings } from "./BarChartForm";
+import { LineChartSettings } from "./LineChartForm";
+
+export type VizType = "table" | "bar" | "line";
+
+export type VizConfigBase = {
+  cachedXY?: { xAxisKey?: string; yAxisKey?: string };
+};
 
 export type VizConfig =
-  | {
-      type: "table";
-      settings: undefined;
-    }
-  | {
-      type: "bar";
-      settings: BarChartSettings;
-    };
+  | ({ type: "table"; settings: undefined } & VizConfigBase)
```

Comment:
Why are we included cachedXY in table? If it's never used in "table" then it's incorrect to truly consider this a part of VizConfigBase. The idea of a "Base" type convention is that all attributes in that Base type will be used. If the table won't use it, then it means cachedXY is specific to charts with XY axes. I can imagine many other visualization types that won't use it (table, maps, gantt charts, sankey diagrams, etc.)